include/zephyr/net/ethernet.h

```diff
@@ -1296,6 +1299,16 @@ int net_eth_txinjection_mode(struct net_if *iface, bool enable);
  */
 int net_eth_mac_filter(struct net_if *iface, struct net_eth_addr *mac,
 		       enum ethernet_filter_type type, bool enable);
+
+/**
+ * @brief Return the PHY device that is tied to this ethernet network interface.
+ *
+ * @param iface Network interface
+ *
+ * @return Pointer to PHY device if found, NULL if not found.
+ */
+const struct device *net_eth_get_phy(struct net_if *iface);
```

Comment:
perhaps, this could return the iface->if_dev if the phy is integrated as part of the interface device itself as a way to signal that information?

Reply:
When using this function I expect to get a device (`const struct device *`) back and not a `struct net_if_dev`. Also I want to be able to directly use this device with function from the ethernet phy api. IMO the ethernet driver would be responsible for providing it. Also zephyr currently only supports providing one api per device.

What would work is to have a separate `DEVICE_DEFINE` for the phy part, like we do for ptp clock.

Reply:
well, I meant iface->if_dev->dev

I will say though I was not too convinced of my own idea here anyways, it would in the api contract to require checking if the device returned is the same device put as the argument, which is a little strange

Reply:
It might make sense though to do what you said for these types of devices that have an integrated mac with phy, to define another device for the phy api, although I don't think the phy API is meant for that necessarily, I don't know if these types of devices have "phy registers" in the same sense that the phy api expects

Reply:
> I don't know if these types of devices have "phy registers" in the same sense that the phy api expects

Usually, yes they do have “similar” registers, except that are memory mapped instead of accessed through mdio.

Reply:
Yes, I wouldn't expect an on chip register to be access by MDIO, or SPI or I2C or any of the other buses that external phys tend to reside on, I'm just wondering maybe it's not a safe assumption that every integrated device will be able to meet this phy API in a reasonable way, and I am actually slightly uncomfortable with the idea I proposed, it was just an idea. Besides, I would not expect this PR to implement the phy API for these macs so it wouldn't be a very useful change to the PR anyways. And the alternative is to not assume that it would meet the phy API and instead require the client to check if the device returned is the same as the one passed in, which is just a strange contract to have.

So, I put my idea out there for consideration but to be clear I actually agree with the current state of just returning NULL in these cases, instead of overcomplicating it, at least for now. Someone owning one of those integrated mac drivers can decide if they want to make use of this API somehow later.